tuplex/core/src/logical/CacheOperator.cc

```diff
@@ -22,19 +22,16 @@ namespace tuplex {
         LogicalOperator::copyMembers(other);
         auto cop = (CacheOperator*)other;
         setSchema(other->getOutputSchema());
-        _normalCasePartitions = cop->cachedPartitions();
-        _generalCasePartitions = cop->cachedExceptions();
-        _partitionToExceptionsMap = cop->partitionToExceptionsMap();
-        // copy python objects and incref for each!
-        _py_objects = cop->_py_objects;
-        python::lockGIL();
-        for(auto obj : _py_objects)
-            Py_XINCREF(obj);
-        python::unlockGIL();
+        _normalPartitions = cop->cachedNormalPartitions();
```

Comment:
hm maybe we should add a struct which contains all of these results... -> would make the code more readable.

Reply:
i.e., normalPartitions + generalPartitions + fallbackPartitions + partitionGroups. Maybe DatasetPartitions? or PartitionCollection? open for good name suggestions here...

Reply:
that struct/class should also contain counts etc.